internal/neofs/tree.go

```diff
@@ -242,6 +252,96 @@ func (c *TreeClient) GetLatestVersion(ctx context.Context, cnrID *cid.ID, object
 	return c.getLatestVersion(ctx, cnrID, versionTree, fileNameKV, path, meta)
 }
 
+func (c *TreeClient) GetLatestVersionsByPrefix(ctx context.Context, cnrID *cid.ID, prefix string) ([]oid.ID, error) {
+	var rootID uint64
+	path := strings.Split(prefix, separator)
+	tailPrefix := path[len(path)-1]
+
+	if len(path) > 1 {
+		meta := []string{fileNameKV}
+
+		nodes, err := c.getNodes(ctx, cnrID, versionTree, fileNameKV, path[:len(path)-1], meta, true)
+		if err != nil {
+			return nil, err
+		}
+		if len(nodes) == 0 {
+			return nil, nil
+		}
+		if len(nodes) != 1 {
+			return nil, layer.ErrNodeNotFound
+		}
+
+		rootID = nodes[0].NodeId
+	}
+
+	subTree, err := c.getSubTree(ctx, cnrID, versionTree, rootID, 1)
+	if err != nil {
+		return nil, err
+	}
+
+	var result []oid.ID
+	for _, node := range subTree {
+		if node.GetNodeId() != 0 && hasPrefix(node, tailPrefix) {
+			latestNodes, err := c.getSubTreeLatestVersions(ctx, cnrID, node.GetNodeId())
```

Comment:
I suppose we can drop `latestNodes` and assign directly to `result`.

Reply:
Could you explain what do you mean? The `result` is slice of `[]oid.ID` but `getSubTreeLatestVersions` returns slice of `*data.NodeVersion`. Besides, we have to append to the `results` rather than replace it (we cannot do this because of the functions returns `error` as second value)

Reply:
Sorry, you're right